aiidalab_widgets_base/computational_resources.py

```diff
@@ -421,12 +421,17 @@ def _can_login(self):
         return ret == 0
 
     def _is_in_config(self):
-        """Check if the config file contains host information."""
+        """Check if the SSH config file contains host information."""
         fpath = Path.home() / ".ssh" / "config"
         if not fpath.exists():
             return False
-        cfglines = open(fpath).read().split("\n")
-        return "Host " + self.hostname.value in cfglines
+        sshcfg = parse_sshconfig(self.hostname.value)
+        # NOTE: parse_sshconfig returns a dict with a hostname
+        # even if it is not in the config file.
+        # We require at least the user to be specified.
+        if "user" not in sshcfg:
```

Comment:
Maybe it is worth setting the `self.message` with the warning.

Reply:
I don't think so. This function is meant to check if the hostname is in config, but it is not an error if it is not there, since the user is typically setting it up when this function is called.

Reply:
I'll let @yakutovicha have another look at it, all good for me.